### ISSUE TITLE
SSM: `wait_until :command_executed` fails with `InvocationDoesNotExist` error

### ISSUE THREAD

**Issue by haines** [aws/aws-sdk-ruby#2556]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide for v3](https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-ruby) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-ruby/issues) and didn't find any solution
  
**Describe the bug**

Calling `Aws::SSM::Client#wait_until :command_executed` immediately after `#send_command` often raises `Aws::SSM::Errors::InvocationDoesNotExist`, presumably due to eventual consistency on the API side. Inserting a sleep between `send_command` and the waiter makes it less likely, but a robust solution essentially requires manually implementing a waiter for the command invocation to exist before waiting for it to be executed.

**Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version**

aws-sdk-ssm 1.111.0

**Version of Ruby, OS environment**

ruby 2.7.3p183 (2021-04-05 revision 6847ee089d) [x86_64-darwin20]

**To Reproduce (observed behavior)**

```ruby
ssm = Aws::SSM::Client.new

instance_id = "..."

send_command_result = ssm.send_command(document_name: "...", instance_ids: [instance_id])
command_id = send_command_result.command.command_id
p ssm.wait_until(:command_executed, command_id: command_id, instance_id: instance_id)
# => ~/.gem/ruby/2.7.3/gems/aws-sdk-core-3.114.2/lib/aws-sdk-core/waiters/waiter.rb:114:in `block in poll': stopped waiting due to an unexpected error: Aws::SSM::Errors::InvocationDoesNotExist (Aws::Waiters::Errors::UnexpectedError)

```

**Expected behavior**

The `command_executed` waiter waits until the command has been executed.

**Additional context**

Related: https://github.com/aws/aws-sdk-js/issues/3457

**Comment by mullermp** [comment]
Thanks for reporting this. You are correct that a "CommandExists" waiter should probably be added. Another alternative might be to extend the current waiter to retry on InvocationDoesNotExist. I'll have to think that one through.

If I were to get you a change to waiters for you to apply locally, would you mind help testing that?

**Comment by haines** [comment]
Yep, happy to try out whatever solution you come up with!

**Comment by alextwoods** [comment]
I've created a PR (#2559) which adds a retry on on the `InvocationDoesNotExistError` - I've tested this locally and it fixes the issue for me.  @haines  - can you test with this change as well?   You can pull the branch and then use `gems/aws-sdk-resources/bin/aws-v3.rb` to launch an interactive REPL with the SDK with the changes.

**Comment by haines** [comment]
@alextwoods I dropped your branch into my Gemfile and ran my application after commenting out the ad-hoc initial wait for the command invocation to exist, and it worked perfectly - thanks!

**Comment by alextwoods** [comment]
Just following up on this - I've got the waiter changes merged into the service teams upstream repository and I'm working with them to get the changes included in an upcoming release.

**Comment by alextwoods** [comment]
The Waiter changes were released today in `aws-sdk-ssm` version `1.118.0`.
